internal/controllers/upgrade_handler.go

```diff
@@ -0,0 +1,157 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/coreos/go-semver/semver"
+	ibgu "github.com/openshift-kni/cluster-group-upgrades-operator/pkg/api/imagebasedgroupupgrades/v1alpha1"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	siteconfig "github.com/stolostron/siteconfig/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func (t *provisioningRequestReconcilerTask) IsUpgradeRequested(
```

Comment:
Maybe add a brief comment describing that if release version in the template is greater than the current ocp version, an upgrade will be triggered